### ISSUE TITLE
Quote Replies: Option to reply as a reblog

### ISSUE THREAD

**Issue by marcustyphoon** [AprilSylph/XKit-Rewritten#1414]
### Concept

Putting this here so I don't forget to look into it at some point.

Sometimes you want to quote reply in a reblog of the post replied to rather than as a new post. This currently isn't supported by quote replies. It's _probably_ possible? If there's a long reblog thread that gets a reply there's some ambiguity around where a reply-as-reblog should start from in the chain.

Sure, one could argue that you can just reply in the replies in this case, but that's true of every use of quote replies too.

**Comment by marcustyphoon** [comment]
> If there's a long reblog thread that gets a reply there's some ambiguity around where a reply-as-reblog should start from in the chain.

Finally looking into this, and indeed, it's not very simple. In activity notifications, we can (mostly?) only access the root post. If something like #2112 is merged, we have a choice between reblogging the root post and reblogging the thread from the point the user is currently viewing. Unless, of course, the root post is deleted, in which case we only have the latter option.

I don't mind coding UI to let the user select which they prefer when there are multiple options, but it makes the test case matrix sort of absurdly huge. (Well, it was already going to be, but.)

It would be kind of cute to use the Trim Reblogs UI to preview the result options, now that I think of it...

**Comment by marcustyphoon** [comment]
> Ugh. Nevermind, I thought all replies on reblogs got sent to the reblogger's activity feed as activity on that reblog, but no, it's only when you don't also own the root post. Ownership of the root post takes precedence. That's... really annoying.
> 
> _Originally posted by @AprilSylph in https://github.com/AprilSylph/XKit-Rewritten/issues/1542#issuecomment-2230743771_

Note: I made a reply on a reblog I didn't own and I'm still seeing the notification linking to the root post, not the reblog. So I don't think it ever does?

**Comment by AprilSylph** [comment]
It definitely did when I tested it back then. Let me try it again.

**Comment by AprilSylph** [comment]
Yep, it still works like this.

- Root post: https://www.tumblr.com/aprilsylph/810239349390229504/this-is-the-root-post
- Reblogged post: https://www.tumblr.com/april-codes/810239430893977600/this-is-a-reblog-by-april-i-dont-own-the-root

I get a reply on the reblogged post. In `@april-codes`'s activity feed, the notification links to the reply on the reblog (https://www.tumblr.com/april-codes/810239430893977600/this-is-a-reblog-by-april-i-dont-own-the-root/replies/810239471608119296), meanwhile in `@aprilsylph`'s activity feed, the notification links to the reply on the root post (https://www.tumblr.com/aprilsylph/810239349390229504/this-is-the-root-post/replies/810239471608119296).

If `@april-codes` had also authored the root post, there would only be one notification since the same blog would then the root post and the reblog. For other activity types (likes, reblogs), any notifications on the reblog would link to the reblog—but for replies, the root takes precedence, and the fact that the reply was made on the reblog at all is completely obscured in the notification.

I actually raised this discrepancy as a bug, but I assume it's still not fixed. I guess I should retest that, too.

---

Yep, still not fixed! If I get a reply on https://www.tumblr.com/aprilsylph/810239944703524864/this-is-a-reblog-of-the-root-post-on-the-same-blog, `@aprilsylph` gets a notification which links to https://www.tumblr.com/aprilsylph/810239349390229504/this-is-the-root-post/replies/810239956029210624.

**Comment by marcustyphoon** [comment]
Oh, right: you tested getting a reply/comment to **your** reblog; I tested one of my blogs being mentioned in a reply/comment on a reblog which is not mine.

**Comment by marcustyphoon** [comment]
I can see the discrepancy you mentioned being intentional? You're saying, right, that if someone replies to a reblog (not by you) of a post (yours), the activity item should link to the reblog (not by you)? I can see the argument that activity items should always link to something you own, as your ownership is why you're getting the notification in the first place.

(This, of course, doesn't apply to the "getting mentioned; neither the reblog or the root post is yours" scenario I tested, and presumably also to the essentially identical "having a threaded reply/comment added to your reply/comment on a reblog (not yours) of a post (not yours)" scenario. You own neither the root post or the reblog, so it would be nice if the correct one was identified.)

Ah, wait, now hold on: what happens in your tested scenario if it's a threaded reply/comment? Do you get notifications for those at all? Does it matter if the parent reply/comment was on the root post or the reblog?

**Comment by AprilSylph** [comment]
> You're saying, right, that if someone replies to a reblog (not by you) of a post (yours), the activity item should link to the reblog (not by you)?

That's not what I'm saying at all! Ugh, the fact that this bug is hard to parse when you're not on the receiving end of it is probably why it still hasn't been fixed.

If someone replies to a reblog, by you, of a post, also by you, the activity item should link to the reblog, by you.

But it doesn't.

Author | Reblogged by | Activity for root | Activity for reblog
-|-|-|-
You | Not you | Your post | Their reblog
Not you | You | Their post | Your reblog
You | You | Your post | Doesn't exist ‼️

**Comment by marcustyphoon** [comment]
Oh, gotcha. Yeah, that's definitely a bug.

**Comment by marcustyphoon** [comment]
> Ah, wait, now hold on: what happens in your tested scenario if it's a threaded reply/comment? Do you get notifications for those at all? Does it matter if the parent reply/comment was on the root post or the reblog?

I do get a notification for a comment made on my reblog that is replying to a comment that I made on the root post, which is not mine, but it links to the root post. 

I do **not** get a notification for a comment made on the root post, which is not mine, that is replying to a comment that I made on said the root post.

I do get a notification for a comment made on my reblog that is replying to a comment that I did not make, on my reblog, and it links to my reblog.

So that means... that... I... wait. Nope. Hold on. Wait. I would need to test—

`if someone replies to [your comment/a comment you didn't make] [/and it mentions you] that you made on [a reblog by you/a reblog by someone else] of [a post by you/a post by someone else] on [the root post/the reblog], then: [you get a notification and it links to the reblog/you get a notification and it links to the root post/you get nothing]`

~32 test cases! Never mind, I give up.

**Comment by marcustyphoon** [comment]
> It would be kind of cute to use the Trim Reblogs UI to preview the result options, now that I think of it...

Look: my philosophy is that you can always **remove** code you've written. Does it make me make stuff like the Find Inactives rewrite that are too long to review? Well, sure.

<img width="600" src="https://github.com/user-attachments/assets/cc85523d-b9b3-4792-a7ca-3614a03c6140" />

https://github.com/AprilSylph/XKit-Rewritten/compare/master...marcustyphoon:XKit-Rewritten:quote-replies-reblog-3-full
